src/app/login/login.component.ts

```diff
@@ -7,9 +8,47 @@ import { Component, OnInit } from '@angular/core';
 })
 export class LoginComponent implements OnInit {
 
+  loginForm: FormGroup;
```

Comment:
@Yuriy-27 I think you shouldn't store all FormConrols as properties of the component.
Store only form (`loginForm`), after you create form you can access to controls like `loginForm.controls.email` or `loginForm.get('email')`.

By the way please google and learn base of FormBuilder in reactive forms. It looks like more convenient to use it for form creation (form building).